Library1/Library.fs

```diff
@@ -22,9 +22,389 @@ module FetchData =
 
     [<CompiledName("FetchForecastsAsync")>]
     let fetchForecasts (http:HttpClient) =
-        //http.GetJsonAsync<WeatherForecast[]>("/sample-data/weather.json")
         async {
-            let! forecasts = Async.AwaitTask <| http.GetJsonAsync<WeatherForecast[]>("/sample-data/weather.json")
-            return forecasts
+            return! Async.AwaitTask <| http.GetJsonAsync<WeatherForecast[]>("/sample-data/weather.json")
         }
         |> Async.StartAsTask
+
+namespace BlazorApp1
+
+open System
+open System.Collections.Generic
+open System.Linq
+open System.Threading.Tasks
+open System.Net.Http
+open Microsoft.AspNetCore.Blazor
+open Microsoft.AspNetCore.Blazor.Components
+open Microsoft.AspNetCore.Blazor.Layouts
+open Microsoft.AspNetCore.Blazor.Routing
+
+// NOTE: _ViewImports.cshtml appear to be required in the BlazorApp.
+type _ViewImports () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+
+namespace BlazorApp1.Shared
+
+open System
+open System.Collections.Generic
+open System.Linq
+open System.Threading.Tasks
+open System.Net.Http
+open Microsoft.AspNetCore.Blazor
+open Microsoft.AspNetCore.Blazor.Components
+open Microsoft.AspNetCore.Blazor.Layouts
+open Microsoft.AspNetCore.Blazor.Routing
+open BlazorApp1
+
+type NavMenu () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+        builder.OpenElement(0, "div")
+        builder.AddAttribute(1, "class", "main-nav")
+        builder.AddContent(2, "\n    ")
+        builder.OpenElement(3, "div")

```

Comment:
Most of this appears fairly straight-forward. I think a DSL could generate this fairly easily.